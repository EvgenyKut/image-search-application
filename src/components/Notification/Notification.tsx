import Snackbar from '@mui/material/Snackbar';
import Alert from '@mui/material/Alert';
import { useAppDispatch, useAppSelector } from '../../hooks';
import { resetNotifications } from '../../store/reducers/NotificationsSlice';

const Notification = () => {
  const dispatch = useAppDispatch();

  const { message, type, open } = useAppSelector(
    (state) => state.notificationReducer,
  );

  const handleClose = () => {
    dispatch(resetNotifications());
  };

  return (
    <Snackbar open={open} autoHideDuration={4000} onClose={handleClose}>
      <Alert
        onClose={handleClose}
        severity={type || 'info'}
        sx={{ width: '100%' }}
      >
        {message || ''}
      </Alert>
    </Snackbar>
  );
};

export default Notification;
