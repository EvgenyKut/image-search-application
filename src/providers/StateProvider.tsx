import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { setupStore } from '../store/store';

type StateProviderProps = {
  children: ReactElement;
};
const StateProvider: React.FC<StateProviderProps> = ({ children }) => {
  const store = setupStore();
  return <Provider store={store}>{children}</Provider>;
};

export default StateProvider;
