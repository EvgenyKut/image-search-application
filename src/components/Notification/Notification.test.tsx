import renderer from 'react-test-renderer';
import StateProvider from '../../providers/StateProvider';
import Notification from './index';

it('renders correctly Notification', () => {
  const tree = renderer
    .create(
      <StateProvider>
        <Notification />
      </StateProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
