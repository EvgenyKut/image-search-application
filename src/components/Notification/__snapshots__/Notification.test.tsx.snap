// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Notification 1`] = `null`;
